base/MAPL_LatLonGridFactory.F90

```diff
@@ -1387,60 +1389,96 @@ subroutine initialize_from_esmf_distGrid(this, dist_grid, lon_array, lat_array,
 
    end subroutine initialize_from_esmf_distGrid
 
+   logical function test_decomp_equals(this,a)
+      class (LatLonGridFactory), intent(in) :: this
+      class (AbstractGridFactory), intent(in) :: a
 
-
-   logical function equals(a, b)
-      class (LatLonGridFactory), intent(in) :: a
-      class (AbstractGridFactory), intent(in) :: b
-
-      select type (b)
+      select type (a)
          class default
-         equals = .false.
+         test_decomp_equals = .false.
          return
       class is (LatLonGridFactory)
-         equals = .true.
+         test_decomp_equals = .true.
 
-         equals = (a%im_world == b%im_world) .and. (a%jm_world == b%jm_world)
-         if (.not. equals) return
-
-         equals = (a%lm == b%lm)
-         if (.not. equals) return
 
-         equals = size(a%ims)==size(b%ims) .and. size(a%jms)==size(b%jms)
-         if (.not. equals) return
+         test_decomp_equals = size(a%ims)==size(this%ims) .and. size(a%jms)==size(this%jms)
+         if (.not. test_decomp_equals) return
 
          ! same decomposition
-         equals = all(a%ims == b%ims) .and. all(a%jms == b%jms)
-         if (.not. equals) return
+         test_decomp_equals = all(a%ims == this%ims) .and. all(a%jms == this%jms)
+         if (.not. test_decomp_equals) return
 
-         equals = (a%is_regular .eqv. b%is_regular)
-         if (.not. equals) return
+      end select
+
+   end function test_decomp_equals
+
+
+   logical function test_physical_params_equals(this, a)
```

Comment:
Again - avoid `test_`.    Maybe `physical_params_are_equal()`

Reply:
Or variant `are_physical_params_equal()`

Reply:
changed to physical_params_are_equal